client/src/components/editBlog/edit.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import FormErrors from '../errors/formErrors'
+import './edit.css';
+
+class Edit extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      blog: {
+        title: '',
+        author: '',
+        body: ''
+      },
+      formErrors: { title: '', author: '', body: '' },
+      titleValid: true,
+      authorValid: true,
+      bodyValid: true,
+      formValid: true
+    };
+  }
+
+  componentDidMount() {
+    fetch('/blogs/' + this.props.match.params.id)
+      .then(res => res.json())
+      .then(blog => {
+        this.setState({ blog });
+      });
+  }
+
+  onChange = (e) => {
+    const name = e.target.name;
+    const value = e.target.value;
+    const state = this.state.blog
+    state[name] = value;
+    this.setState({ blog: state }, () => { this.validateField(name, value) });
+  }
+
+  onSubmit = (e) => {
+    e.preventDefault();
+
+    const { title, author, body } = this.state.blog;
+
+    fetch('/blogs/' + this.props.match.params.id, {
+      method: 'PUT',
+      mode: 'CORS',
+      body: JSON.stringify({ title, author, body }),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    }).then(res => {
+      this.props.history.push("/blog/" + this.props.match.params.id)
+    }).catch(err => err);
+  }
+
+  validateField(fieldName, value) {
+    let fieldValidationErrors = this.state.formErrors;
+    let titleValid = this.state.titleValid;
+    let authorValid = this.state.authorValid;
+    let bodyValid = this.state.bodyValid;
+
+    switch (fieldName) {
+      case 'title':
+        titleValid = value.length >= 3 && value.length <= 100;
```

Comment:
Magic number 3 and 100?
Define constants, please

Reply:
done